### PR TITLE
Adding new document about how to migrate Osclass to docker infrastruc…

### PR DESCRIPTION
**Pull request by XaviOutside**
…ture with Docker Compose.

### CODE REVIEW

**Comment by XaviOutside** [comment]
Thanks Gordon,

I’m already updated.

Best,
Xavi.

**Comment by SvenDowideit** [comment]
mmm, i suspect this should go into the http://github.com/docker/compose repository documentation.

@bfirsh ?

**Comment by crosbymichael** [comment]
ping @moxiegirl

File: docs/sources/examples/getting_started_with_compose_and_osclass.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+page_title: Getting started with compose and Osclass
```
Comment (moxiegirl) [inline docs/sources/examples/getting_started_with_compose_and_osclass.md]: Very nice contribution @XaviOutside.  Some of the structure in your documentation is non-standard and uses non-English phrasing.  So...

---> https://gist.github.com/moxiegirl/86ef1eb7320bf03133ac

page_title: Getting started with Docker Compose and Osclass
page_description: Migrating and running an Osclass installation.
page_keywords: docker, example, package installation, osclass, compose

# Getting started with Docker Compose and Osclass

This document teaches you how to migrate an Osclass installation to a Docker
infrastructure. When you are done, your Docker infrastructure will contain six Docker containers, three  
server containers and three volume containers:

* Mysql database
* Postfix server
* Apache2 Webserver
* Osclass volume 
* Database volume
* Backup volume

This configuration was tested on Osclass 3.5.

## Prerequisites

Before you begin, make sure you have installed the following software:

* VirtualBox
* Docker
* Docker Machine 
* Docker compose 

Refer to the appropriated Docker documentation for different distributions for installing.

## Set up the servers

1. Create the virtualbox machine.

        $ docker-machine create --driver virtualbox dockerenv 

        $ docker-machine active dockerenv

        $ $(docker-machine env dockerenv)

2. Download the scripts using `git clone`.

        $ git clone https://github.com/XaviOutside/MIGRATION_OSCLASS_2_DOCKER.git
        
3. Change to the folder directory.

        $ cd MIGRATION_OSCLASS_2_DOCKER

4. Open the  in `common.env` file in your favorite editor.

        $ vi common.env 


5. Set the `DOMAIN` and `MYSQL_ROOT_PASSWORD` to contain your domain and password.

         DOMAIN=yourdomain.org
         MYSQL_ROOT_PASSWORD=yourpassword

6. Save and close the file.

7. Build containers.

        $ docker-compose build

8. Run containers.

        $ docker-compose up

9. Launch the `http://your_ip_docker_machine` URL in your favorite browses.

    If you don't know it, you can get your `ip` with the following command:

        $ docker-machine ip
     

## Import data

1. Create a `tar.gz` file of your current Osclass server.

2. Replace the `backup_osclass.tar.gz` in the `migration` folder.
     
3. Create a dump file of your Osclass database.

4. Replace the `backup.mysql.sql` in the `migration` folder.
     
5. Copy the `migration` folder into the Osclass container.

        $ tar -cf - migration | docker exec -i <NAME_OSCLASS_CONTAINER>  /bin/tar -C / -xf -

6. Run the `osclass_init.sh` script to import the files and database data into the volume containers.

        $ docker exec -it <NAME_OSCLASS_CONTAINER> bash /osclass_init.sh

## Backups

Generate a backup of our osclass files and database.

    $ docker exec -it <NAME_OSCLASS_CONTAINER> bash /osclass_backup.sh

## Others

There is an `INSTALL.sh` file if you want to launch without `docker-compose`.

**Comment by moxiegirl** [comment]
@XaviOutside HI Xavier --- I see you made your contribution! Very cool, thanks for doing this. Hopefully you are enjoying Spain as it is warmer than San Francisco.  Maybe we will see you at the Docker Party next year in Spain?

**Comment by calavera** [comment]
I think this should go in https://github.com/docker/compose, as @SvenDowideit mentions.

**Comment by XaviOutside** [comment]
@moxiegirl
Hi moxiegirl!

Sorry for the delay but I have moved to Berlin (June) and I have started a new job in a small startup. I have been disconnected of the world because I haven’t had enough time and a good internet connection too. Now, I’m in my new apartment with DSL, I will be able to review the different things that you have send to me.
I have seen that the next Dockercon is in Barcelona! Good news, I don’t know if I will be able to go but anyway, I’m very happy that Docker comes to BCN.
I’m working as a devops and I’m working with docker mainly. We are using dockerhub in order to store the images and I have more ideas to contribute.

Regards,
Xavier.

**Comment by XaviOutside** [comment]
Ok, I'm going to move this doc to https://github.com/docker/compose.

**Comment by thaJeztah** [comment]
Thanks @XaviOutside! I see you created https://github.com/docker/compose/pull/1627 so I'll close this PR

(I hope you enjoy Berlin by the way, I love that city!)

**Pull request closed without merging**